libevm/JitVM.cpp

```diff
@@ -14,11 +13,135 @@ namespace dev
 {
 namespace eth
 {
+namespace
+{
+
+static_assert(sizeof(Address) == sizeof(evm_hash160),
+              "Address types size mismatch");
+static_assert(alignof(Address) == alignof(evm_hash160),
+              "Address types alignment mismatch");
+
+evm_hash160 toEvmC(Address _addr)
+{
+	return *reinterpret_cast<evm_hash160*>(&_addr);
+}
+
+Address fromEvmC(evm_hash160 _addr)
+{
+	return *reinterpret_cast<Address*>(&_addr);
+}
+
+static_assert(sizeof(h256) == sizeof(evm_hash256),
+              "Hash types size mismatch");
+static_assert(alignof(h256) != alignof(evm_hash256),
+              "Hash types alignment match -- update implementation");
+
+evm_hash256 toEvmC(h256 _h)
+{
+	// Because the h256 only pretends to be aligned to 8 bytes we have to do
+	// non-aligned memory copy.
+	// TODO: h256 should be aligned to 8 bytes.
+	evm_hash256 g;
+	std::memcpy(&g, &_h, sizeof(g));
+	return g;
+}
+
+evm_uint256 toEvmC(u256 _n)
+{
+	evm_uint256 m;
+	m.words[0] = static_cast<uint64_t>(_n);
+	_n >>= 64;
+	m.words[1] = static_cast<uint64_t>(_n);
+	_n >>= 64;
+	m.words[2] = static_cast<uint64_t>(_n);
+	_n >>= 64;
+	m.words[3] = static_cast<uint64_t>(_n);
+	return m;
+}
 
-extern "C" void env_sload(); // fake declaration for linker symbol stripping workaround, see a call below
+u256 fromEvmC(evm_uint256 _n)
+{
+	u256 u = _n.words[3];
+	u <<= 64;
+	u |= _n.words[2];
+	u <<= 64;
+	u |= _n.words[1];
+	u <<= 64;
+	u |= _n.words[0];
+	return u;
+}
+
+evm_variant evm_query(evm_env* _opaqueEnv, evm_query_key _key,
```

Comment:
I changed that but I think we should give up formatting that cannot be done by any tool. I tried to force clang-format to style like that, no success so far :)